rdkit/Chem/Features/ShowFeats.py

```diff
@@ -169,13 +169,22 @@ def ShowMolFeats(mol, factory, viewer, radius=0.5, confId=-1, showOnly=True, nam
     dirLabel = featLabel + "-dirs"
     viewer.server.resetCGO(dirLabel)
 
-  for i, feat in enumerate(molFeats):
+  FeatVectsDictMethod = \
+    {('Aromatic', None): FeatDirUtils.GetAromaticFeatVects, 
+     ('Donor', 1): FeatDirUtils.GetDonor1FeatVects,
+     ('Donor', 2): FeatDirUtils.GetDonor2FeatVects,
+     ('Donor', 3): FeatDirUtils.GetDonor3FeatVects,
+     ('Acceptor', 1): FeatDirUtils.GetDonor1FeatVects,
+     ('Acceptor', 2): FeatDirUtils.GetDonor2FeatVects,
+     ('Acceptor', 3): FeatDirUtils.GetDonor3FeatVects,
+     } # TODO: add more feat types if needed. Error should not be thrown if a type is not found due to the lack of method.
```

Comment:
I can see why you do this and it reduces the code for the donors and acceptors. On the one hand that is nice, on the other hand however, we loose the clarity of which functions are called when the dictionary is used to select the method. A alternative would be to keep the original call for the aromatic feature and use two dictionaries one for the donor and one for the acceptor feature. These can be defined in the respective if block.

Reply:
Agree ! I would fix it in the next patch. Thank you for your review.